server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java

```diff
@@ -117,6 +131,12 @@ public void writeFile(StoreFileMetadata fileMetadata, long readSnapshotFileBuffe
         }
     }
 
+    private IndexOutput createIndexOutput(String tempFileName, StoreFileMetadata fileMetadata) throws IOException {
+        return verifyOutput
+            ? store.createVerifyingOutput(tempFileName, fileMetadata, IOContext.DEFAULT)
+            : store.directory().createOutput(tempFileName, IOContext.DEFAULT);
```

Comment:
We will not need `verifyOutput` in the future. Alternatively I could make `createIndexOutput` protected and override it in the plugin.